src/main/resources/templates/index.html

```diff
@@ -15,11 +15,101 @@ <h1 class="text-center">
     Create personalized short-links
   </p>
 
-  <div class="text-center m-5">
-    <p>
-      You need to be an admin to create new short links.
-    </p>
-    <a class="btn btn-primary" th:href="@{/admin}">Admin Panel</a>
+  <div class="alert alert-warning" id="error-message" th:if="${error} != null" th:text="${error}">
+    Some error message
+  </div>
+
+  <div class="alert alert-success" id="success-message" th:if="${success} != null" th:text="${success}">
+    Some success message
+  </div>
+
+  <div sec:authorize="isAnonymous()" class="alert alert-info" role="alert">
+    To access the admin panel, you must first login.
+  </div>
+
+  <div sec:authorize="isAnonymous()">
+    <form th:action="@{/login}" method="post">
+      <div class="form-group">
+        <label for="username">Username</label>
+        <input type="text" id="username" name="username" class="form-control"/>
+      </div>
+      <div class="form-group">
+        <label for="password">Password</label>
+        <input type="password" class="form-control" id="password" name="password"/>
+      </div>
+      <input type="hidden"
+             name="${_csrf.parameterName}"
+             value="${_csrf.token}"/>
```

Comment:
Thymeleaf should add these csrf fields on its own, provided you use some types of th: tags(like th:action) in the form tag.